### ISSUE TITLE
False negatives for variable_name rule.

### ISSUE THREAD

**Issue by nickmshelley** [realm/SwiftLint#549]
If I have a file that contains the following, only `thing1_`, `Thing_`, and `thing2_` are caught, but `thingFunc_` and `thing3_` are not.

``` swift
let thing1_ = "hi"

struct Thing_ {
    let thing2_: String

    func thingFunc_() {
        let thing3_ = "hi"
        print(thing3_)
    }
}
```

**Comment by norio-nomura** [comment]
It seems SourceKitService does not return structure about `let thing3_ = "hi"`.
```sh
% sourcekitten structure --file 549.swift
{
  "key.substructure" : [
    {
      "key.kind" : "source.lang.swift.decl.var.global",
      "key.offset" : 0,
      "key.nameoffset" : 4,
      "key.namelength" : 7,
      "key.length" : 18,
      "key.accessibility" : "source.lang.swift.accessibility.internal",
      "key.typename" : "thing1_",
      "key.name" : "thing1_"
    },
    {
      "key.kind" : "source.lang.swift.decl.struct",
      "key.offset" : 20,
      "key.nameoffset" : 27,
      "key.namelength" : 6,
      "key.bodyoffset" : 35,
      "key.bodylength" : 106,
      "key.accessibility" : "source.lang.swift.accessibility.internal",
      "key.substructure" : [
        {
          "key.kind" : "source.lang.swift.decl.var.instance",
          "key.offset" : 40,
          "key.nameoffset" : 44,
          "key.namelength" : 7,
          "key.length" : 19,
          "key.accessibility" : "source.lang.swift.accessibility.internal",
          "key.typename" : "String",
          "key.name" : "thing2_"
        },
        {
          "key.kind" : "source.lang.swift.decl.function.method.instance",
          "key.offset" : 65,
          "key.nameoffset" : 70,
          "key.namelength" : 12,
          "key.bodyoffset" : 84,
          "key.bodylength" : 55,
          "key.accessibility" : "source.lang.swift.accessibility.internal",
          "key.substructure" : [
            {
              "key.kind" : "source.lang.swift.expr.call",
              "key.offset" : 120,
              "key.nameoffset" : 120,
              "key.namelength" : 5,
              "key.bodyoffset" : 126,
              "key.bodylength" : 7,
              "key.length" : 14,
              "key.name" : "print"
            }
          ],
          "key.name" : "thingFunc_()",
          "key.length" : 75
        }
      ],
      "key.name" : "Thing_",
      "key.length" : 122
    }
  ],
  "key.offset" : 0,
  "key.diagnostic_stage" : "source.diagnostic.stage.swift.parse",
  "key.length" : 143
}
```
Newer version of SourceKitService in `swift-DEVELOPMENT-SNAPSHOT-2016-02-08-a` has the same behavior.

**Comment by aamctustwo** [comment]
Confirmed still a bug in SwiftLint 0.16

**Comment by marcelofabri** [comment]
Closing as duplicate of #136.
